co-noir/ultrahonk/src/honk_curve.rs

```diff
@@ -0,0 +1,114 @@
+use ark_ec::pairing::Pairing;
+use ark_ff::{One, PrimeField};
+use num_bigint::BigUint;
+
+// Des describes the PrimeField used for the Transcript
+pub trait HonkCurve<Des: PrimeField>: Pairing {
+    const NUM_BASEFIELD_ELEMENTS: usize;
+    const NUM_SCALARFIELD_ELEMENTS: usize;
+
+    fn g1_affine_from_xy(x: Self::BaseField, y: Self::BaseField) -> Self::G1Affine;
+    fn g1_affine_to_xy(p: &Self::G1Affine) -> (Self::BaseField, Self::BaseField);
+
+    fn convert_scalarfield_into(src: &Self::ScalarField) -> Vec<Des>;
```

Comment:
Something like this could work:
 fn convert_scalarfield_into<const SIZE: usize>(src: &ark_bn254::Fr) -> [ark_bn254::Fr; SIZE] {
        assert_eq!(SIZE, Self::NUM_SCALARFIELD_ELEMENTS);
        [src.to_owned(); SIZE]
    }

Reply:
but calling it then is cumbersome as well, since I cannnot use P::NUM_SCALARFIELD_ELEMENTS there either

Reply:
yeah i think this is not necessary atm